Bottom: d79d5464622c07a1ee7c42051fee5fca02c52207
Top:    69d5a78d671e7db0e0da70e3702c92eceac177a8
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-04 13:22:10 -0700

rsocket: Handle TCP_MAXSEG socket option

netperf uses the TCP_MAXSEG socket option.  Add support for it.
Problem reported by Sridhar Samudrala <sri@us.ibm.com>

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/rsocket.c b/src/rsocket.c
index c111797..e906fd4 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1713,6 +1713,9 @@ int rsetsockopt(int socket, int level, int optname,
 			opt_on = *(int *) optval;
 			ret = 0;
 			break;
+		case TCP_MAXSEG:
+			ret = 0;
+			break;
 		default:
 			break;
 		}
@@ -1799,6 +1802,12 @@ int rgetsockopt(int socket, int level, int optname,
 			*((int *) optval) = !!(rs->tcp_opts & (1 << optname));
 			*optlen = sizeof(int);
 			break;
+		case TCP_MAXSEG:
+			*((int *) optval) = (rs->cm_id && rs->cm_id->route.num_paths) ?
+					    1 << (7 + rs->cm_id->route.path_rec->mtu) :
+					    2048;
+			*optlen = sizeof(int);
+			break;
 		default:
 			ret = ENOTSUP;
 			break;
